### PR TITLE
Fix garage-sign deb and added to docker tests

### PR DESCRIPTION
**Pull request by patriotyk**
This PR fixes two issues.
1) garage-deploy.deb first time creates without content from garage-sign.gz
2) It fixes PRO-4473

### CODE REVIEW

File: src/sota_tools/CMakeLists.txt [hunk 1]
```diff
@@ -98,17 +98,16 @@ add_dependencies(build_tests garage-deploy)
 
 
 install(TARGETS garage-deploy RUNTIME DESTINATION bin COMPONENT garage_deploy)
-add_custom_target(download_garage_sign COMMAND wget --no-check-certificate https://ats-tuf-cli-releases.s3-eu-central-1.amazonaws.com/cli-0.2.0-29-gf6f095a.tgz -O ${CMAKE_CURRENT_BINARY_DIR}/garage-sign.tgz
-COMMAND tar -xzf ${CMAKE_CURRENT_BINARY_DIR}/garage-sign.tgz -C ${CMAKE_CURRENT_BINARY_DIR}
-COMMAND chmod +x ${CMAKE_CURRENT_BINARY_DIR}/garage-sign/bin/*)
+execute_process(COMMAND wget --no-check-certificate https://ats-tuf-cli-releases.s3-eu-central-1.amazonaws.com/cli-0.2.0-29-gf6f095a.tgz -O ${CMAKE_CURRENT_BINARY_DIR}/garage-sign.tgz)
```
Comment (patrickvacek) [inline src/sota_tools/CMakeLists.txt]: Can we upgrade this to `https://ats-tuf-cli-releases.s3-eu-central-1.amazonaws.com/cli-0.2.0-52-g83fbfe6.tgz`? There have been a number of important fixes that we should be using.

**Comment by patrickvacek** [comment]
Can we also prevent rebuilding garage-deploy.deb if it already exists and the dependencies haven't changed?

**Comment by patrickvacek** [comment]
I think there's also a problem with the dependencies for the execute_process commands to download garage-sign. For example, if I delete the downloaded files, then I get an error like this:

```
CPack: -   Install component: garage_deploy
CMake Error at /home/patrick/Code/pro/client/aktualizr/build/src/sota_tools/cmake_install.cmake:84 (file):
  file INSTALL cannot find
  "/home/patrick/Code/pro/client/aktualizr/build/src/sota_tools/garage-sign/bin/garage-sign".
Call Stack (most recent call first):
  /home/patrick/Code/pro/client/aktualizr/build/cmake_install.cmake:41 (include)


CMake Error at /home/patrick/Code/pro/client/aktualizr/build/src/sota_tools/cmake_install.cmake:91 (file):
  file INSTALL cannot find
  "/home/patrick/Code/pro/client/aktualizr/build/src/sota_tools/garage-sign/lib/ch.qos.logback.logback-classic-1.2.3.jar".
Call Stack (most recent call first):
  /home/patrick/Code/pro/client/aktualizr/build/cmake_install.cmake:41 (include)
```

**Comment by patriotyk** [comment]
Looks like I have fixed all mentioned problems.

**Comment by patrickvacek** [comment]
Definite improvement, thanks! Do you think it is possible to not rebuild garage-deploy.deb if it already exists and the dependencies haven't changed? I recognize that might be harder, and it's not really a big deal anyway. The downloading was my main concern.

Also, it appears there is now a `cli-0.2.0-54-g1e7a646.tgz`... would you mind bumping that again?

**Comment by cajun-rat** [comment]
I added a dependency to garage-sign for the garage-deploy cpack.  Without this 'ninja qa' was failing with the following error:

~~~
[128/218] cd /home/phil/pro/client/aktualizr/build && cpack
FAILED: src/sota_tools/CMakeFiles/garage-deploy-deb 
cd /home/phil/pro/client/aktualizr/build && cpack
CPack: Create package using DEB
CPack: Install projects
CPack: - Install project: aktualizr
CPack: -   Install component: garage_deploy
CMake Error at /home/phil/pro/client/aktualizr/build/src/sota_tools/cmake_install.cmake:84 (file):
  file INSTALL cannot find
  "/home/phil/pro/client/aktualizr/build/src/sota_tools/garage-sign-prefix/src/garage-sign/bin/garage-sign".
Call Stack (most recent call first):
  /home/phil/pro/client/aktualizr/build/cmake_install.cmake:42 (include)


CMake Error at /home/phil/pro/client/aktualizr/build/src/sota_tools/cmake_install.cmake:88 (file):
  file INSTALL cannot find
  "/home/phil/pro/client/aktualizr/build/src/sota_tools/garage-sign-prefix/src/garage-sign/lib".
Call Stack (most recent call first):
  /home/phil/pro/client/aktualizr/build/cmake_install.cmake:42 (include)


CPack Error: Error when generating package: aktualizr
[137/218] Building CXX object tests/CMakeFiles/t_uptane.dir/uptane_test.cc.o
ninja: build stopped: subcommand failed.
~~~

**Pull request merged**